File: tests/data-conversion/question/question.cpp
#include "runtime/object.h"

namespace {

void TakeNumber(double n);

class LinkNode : public compilets::Object {
 public:
  std::optional<double> item;
  cppgc::Member<LinkNode> next;

  LinkNode(double item) {
    this->item = item;
  }

  void Trace(cppgc::Visitor* visitor) const override {
    compilets::TraceMember(visitor, next);
  }

  virtual ~LinkNode() = default;
};

void TestQuestionTokenInClass() {
  LinkNode* head = compilets::MakeObject<LinkNode>(0);
  if (!head->next) {
    head->next = compilets::MakeObject<LinkNode>(1);
  }
  std::optional<double> i = head->item;
  head->next->item = 3;
  TakeNumber(head->item.value());
  double n = true ? head->item.value() : 0;
  LinkNode* l = true ? head : head->next.Get();
  double memberExam = head->item.value();
  double valueExam = i.value();
}

void TakeNumber(double n) {}

}  // namespace

File: tests/data-conversion/question/question.ts
class LinkNode {
  item?: number;
  next?: LinkNode;

  constructor(item: number) {
    this.item = item;
  }
}

function TestQuestionTokenInClass() {
  const head = new LinkNode(0);
  if (!head.next) {
    head.next = new LinkNode(1);
  }

  let i = head.item;
  head.next.item = 3;
  TakeNumber(head.item!);

  let n: number = true ? head.item! : 0;
  let l: LinkNode = true ? head : head.next!;

  let memberExam = head.item!;
  let valueExam = i!;
}

function TakeNumber(n: number) {
}
